JustSaying.IntegrationTests/Fluent/IntegrationTestBase.cs

```diff
@@ -80,7 +80,17 @@ protected async Task WhenAsync(IServiceCollection services, Func<IMessagePublish
 
             using (var source = new CancellationTokenSource(Timeout))
             {
-                await Task.WhenAny(action(publisher, listener, source.Token), Task.Delay(Timeout, source.Token)).ConfigureAwait(false);
+                var delayTask = Task.Delay(Timeout, source.Token);
+                var actionTask = action(publisher, listener, source.Token);
+
+                await Task.WhenAny(actionTask, delayTask).ConfigureAwait(false);
+
+                source.Token.ThrowIfCancellationRequested();
+
+                if (!actionTask.IsCompletedSuccessfully)
```

Comment:
If this is false because it hasn't finished, then the `throw` might fail. Maybe we need a separate branch in this block which throws with a "still running and hasn't failed" if it's still not done?

Reply:
Then we can use `if (actionTask.IsFaulted)`